src/stock-items/stock-items-table.component.tsx

```diff
@@ -37,6 +43,8 @@ interface StockItemsTableProps {
 
 const StockItemsTableComponent: React.FC<StockItemsTableProps> = () => {
   const { t } = useTranslation();
+  const [searchInput, setSearchInput] = useState("");
+  const searchInputRef = useRef(null);
```

Comment:
@Daphne210, I've tested the solution you provided, and it works as expected. However, I would like to suggest an enhancement to optimize the search functionality further. Instead of triggering an API request for every keystroke, it would be more efficient to implement a debounce mechanism. This technique will ensure that the API request is only fired after the user has stopped typing for a short period, say a few hundred milliseconds

I had done some implementation [here](https://github.com/agesa3/openmrs-esm-stock-management/blob/fx-pagination-search/src/stock-items/stock-items-table.component.tsx)